### PR TITLE
DynamicMethod permit ref returns in .NET Core 2.1+

### PR DESCRIPTION
**Pull request by jkotas**
Fixes https://github.com/dotnet/runtime/issues/92670

### CODE REVIEW

**Review by gewarren — APPROVED** [review]
Looks good in preview: https://review.learn.microsoft.com/en-us/dotnet/api/system.reflection.emit.dynamicmethod.-ctor?view=net-8.0&branch=pr-en-us-9335#system-reflection-emit-dynamicmethod-ctor(system-string-system-type-system-type())

File: xml/System.Reflection.Emit/DynamicMethod.xml [hunk 1]
```diff
@@ -430,8 +430,8 @@
  -or-  
   
  <paramref name="m" /> is <see langword="null" />.</exception>
-        <exception cref="T:System.NotSupportedException">
-          <paramref name="returnType" /> is a type for which <see cref="P:System.Type.IsByRef" /> returns <see langword="true" />.</exception>
+        <exception cref="T:System.NotSupportedException">.NET Framework and .NET Core versions older than 2.1:
+            <paramref name="returnType" /> is a type for which <see cref="P:System.Type.IsByRef" /> returns <see langword="true" />.</exception>
```
Comment (buyaa-n) [inline xml/System.Reflection.Emit/DynamicMethod.xml:434]: NIT: a space

**Review by buyaa-n — APPROVED** [review]
Thanks!

File: xml/System.Reflection.Emit/DynamicMethod.xml (hunk 1 above)
Comment (jkotas) [inline xml/System.Reflection.Emit/DynamicMethod.xml:434]:
```suggestion
          <paramref name="returnType" /> is a type for which <see cref="P:System.Type.IsByRef" /> returns <see langword="true" />.</exception>
```

**Pull request merged**